components/Auth/Auth.js

```diff
@@ -0,0 +1,184 @@
+import { Component } from "react";
+import PropTypes from "prop-types";
+import axios from "axios";
+import get from "lodash/get";
+import { toast } from "react-toastify";
+
+import AuthContext from "../../contexts/AuthContext";
+
+class Auth extends Component {
+  static defaultProps = {
+    children: null,
+  };
+
+  static propTypes = {
+    children: PropTypes.node,
+  };
+
+  state = {
+    authToken: "",
+    user: null,
+    verifyingAuth: true,
+    loadingUserInfo: false,
+  };
+
+  api = axios.create({
+    baseURL: process.env.NEXT_PUBLIC_API_BASE_URL,
+  });
+
+  componentDidMount() {
```

Comment:
I was thinking if we could move authToken check and fetching userInfo to componentWillMount()/constructor()?

Reply:
The downside of keeping it in didMount() will be it will be called after first render which means we need to apply failure/success responses after first render onto DOM, IMO loading the user directly onto the logged in page or non logged in page would be a better UX than causing it to load after first re-render

Reply:
componentWillMount has been deprecated. It is generally recommended to fetch api data in componentDidMount.

Reply:
right correct it's deprecated  and we should call api in didMount.Let's discuss this tomorrow.